db/db_impl_write.cc

```diff
@@ -584,6 +716,38 @@ Status DBImpl::WriteToWAL(const WriteThread::WriteGroup& write_group,
   return status;
 }
 
+Status DBImpl::ConcurrentWriteToWAL(const WriteThread::WriteGroup& write_group,
```

Comment:
It meant to say that this function can be called concurrently. Inside, the actual write to WAL is serialized. I chose this name since previously WriteToWAL assumed to be called only by one thread at a time. ConcurrentWriteToWAL is supposed to suggest that this function can now be called concurrently.